File: HorseRacing.Contracts/Models/Game/Responses/GetWaitingGames/GetWaitingGamesResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.GetWaitingGames
{
    public class GetWaitingGamesResponse : BaseResponse<GetWaitingGamesResponseDto>
    {
        public GetWaitingGamesResponse() : base() { }

        public GetWaitingGamesResponse(GetWaitingGamesResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/get-waiting-games/get-waiting-games-response.ts
import type { BaseResponse } from "~/interfaces/api/contracts/base/responses/base-response";
import type { GetWaitingGamesResponseDto } from "./get-waiting-games-response-dto";

export class GetWaitingGamesResponse implements BaseResponse<GetWaitingGamesResponseDto> {
    Data: GetWaitingGamesResponseDto;

    constructor(data?: GetWaitingGamesResponseDto) {
        this.Data = data || { Games: []};
    }
}
